custom_components/dualmode_generic/climate.py

```diff
@@ -507,7 +513,10 @@ async def async_set_preset_mode(self, preset_mode: str):
         if preset_mode == PRESET_AWAY and not self._is_away:
             self._is_away = True
             self._saved_target_temp = self._target_temp
-            self._target_temp = self._away_temp
+            if self._hvac_mode == HVAC_MODE_COOL:
+                self._target_temp = self._away_temp_cooler
```

Comment:
I think this one just needs a conditional to check for `self._away_temp_cooler` and if it exists, use it, otherwise use `self._away_temp`. Same for the heater conditional underneath.